graphql_compiler/compiler/expressions.py

```diff
@@ -489,8 +505,9 @@ def to_sql(self, aliases, current_alias):
                                       u'process field {}.'.format(self.location.field))
 
         if self.location.field is not None:
-            if '@' in self.location.field:
-                raise NotImplementedError(u'The SQL backend does not support __typename.')
+            if self.location.field == TYPENAME_META_FIELD_NAME:
+                raise NotImplementedError(u'The SQL backend does not support {}.'.format(
+                    TYPENAME_META_FIELD_NAME))
```

Comment:
It might be better to have the following structure:
```
if self.location.field == '__some_field_that_is_supported_on_this_backend':
    implementation_of_field...
elif is_meta_field(self.location.field)
    raise NotImplementedError(u'...')
```

That way new meta fields will be unsupported by default until they are actually implemented. Also there wouldn't be a need to scatter this logic in multiple files: https://github.com/kensho-technologies/graphql-compiler/blob/master/graphql_compiler/compiler/emit_sql.py#L236

Reply:
Updated to the new structure, but I think the `_x_count` check in `emit_sql` may still necessary. We are appending outputs by adding an array aggregation of `self.output_vertex_alias.c[fold_output.field]` to the list of folded outputs. However, this does not work `fold_output.field == _x_count` since a column called `_x_count` does not exist. All of this is done without accessing an expression's `to_sql` method. Does that make sense?

Reply:
Agreed with both of you. We'd probably still need the logic in more than one place, but it's nice to design for "unsupported by default until explicitly supported" whenever it's possible to do that. 🎉 for merging this!